python/tchannel/tornado/util.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import absolute_import
+
+import tornado
+import tornado.gen
+from ..exceptions import TChannelException
+
+
+@tornado.gen.coroutine
+def get_arg(context, index):
+    if index < len(context.argstreams):
+        arg = ""
+        try:
+            chunk = yield context.argstreams[index].read()
+            while chunk:
+                arg += chunk
+                chunk = yield context.argstreams[index].read()
+        finally:
+            raise tornado.gen.Return(arg)
+    else:
+        raise TChannelException()
+
+
+@tornado.gen.coroutine
+def get_all_args(context):
+    arg1 = yield get_arg(context, 0)
+    arg2 = yield get_arg(context, 1)
+    arg3 = yield get_arg(context, 2)
+
+    raise tornado.gen.Return((arg1, arg2, arg3))
+
+
+@tornado.gen.coroutine
+def print_arg(request, index):
```

Comment:
Probably remove this if it's not being used. If it is being used, make it more generic -- something that writes the contents of an arg stream to a file-like object instead of just stdout.